app/models/request.rb

```diff
@@ -21,7 +21,11 @@ class Request < ApplicationRecord
   before_create :set_context
 
   def as_json(options = {})
-    super(options.merge(:methods => [:total_stages, :active_stage]))
+    super(options.merge(:methods => [:total_stages, :active_stage_number]))
+  end
+
+  def current_stage
+    stages.find_by(:state => [Stage::NOTIFIED_STATE, Stage::PENDING_STATE])
```

Comment:
`find_by` is a db operation while `find_index` in the other method is an array operation. Can you guarantee `find_by` returns the first sorted qualified stage?

Reply:
Based on Rails doc, `find_by` will guarantee to return the first matching record based on the condition.